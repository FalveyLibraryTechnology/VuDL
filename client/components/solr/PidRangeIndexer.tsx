import React, { useState } from "react";
import { baseUrl } from "../../util/routes";
import { useFetchContext } from "../../context/FetchContext";
interface PidRangeIndexerProps {
    setResults: (pid: string) => void;
}
const PidRangeIndexer = ({ setResults }: PidRangeIndexerProps): React.ReactElement => {
    const [prefix, setPrefix] = useState("vudl:");
    const [to, setTo] = useState("");
    const [from, setFrom] = useState("");
    const {
        action: { fetchText },
    } = useFetchContext();

    const doApiCall = async () => {
        setResults("Loading...");
        try {
            setResults(
                await fetchText(`${baseUrl}/messenger/queuesolrindex`, {
                    method: "POST",
                    body: JSON.stringify({ prefix, to, from }),
                })
            );
        } catch (error) {
            setResults((error as Error).message);
        }
    };

    return (
        <>
            <label>
                Prefix:
                <input id="pidRangePrefix" type="text" value={prefix} onChange={(e) => setPrefix(e.target.value)} />
            </label>
            <label>
                From (number):
                <input id="pidRangeFrom" type="text" value={from} onChange={(e) => setFrom(e.target.value)} />
            </label>
            <label>
                To (number): <input id="pidRangeTo" type="text" value={to} onChange={(e) => setTo(e.target.value)} />
            </label>
            <button onClick={() => doApiCall()}>Queue Range</button>
        </>
    );
};

export default PidRangeIndexer;
