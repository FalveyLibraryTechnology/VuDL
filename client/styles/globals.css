:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
