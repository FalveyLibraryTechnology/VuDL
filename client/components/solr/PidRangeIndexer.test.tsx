import React from "react";
import { beforeEach, describe, expect, it, jest } from "@jest/globals";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import PidRangeIndexer from "./PidRangeIndexer";

const mockUseFetchContext = jest.fn();
jest.mock("../../context/FetchContext", () => ({
    useFetchContext: () => {
        return mockUseFetchContext();
    },
}));

describe("PidRangeIndexer", () => {
    let fetchContextValues;
    let wrapper;
    let setResults;
    beforeEach(() => {
        fetchContextValues = {
            action: {
                fetchText: jest.fn(),
            },
        };
        mockUseFetchContext.mockReturnValue(fetchContextValues);
        setResults = jest.fn();
        wrapper = mount(<PidRangeIndexer setResults={setResults} />);
    });

    it("renders", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("submits appropriate JSON", async () => {
        fetchContextValues.action.fetchText.mockResolvedValue("testText");

        await act(async () => {
            wrapper.find("#pidRangePrefix").simulate("change", {
                target: {
                    value: "foo:",
                },
            });
            await wrapper.update();
            wrapper.find("#pidRangeFrom").simulate("change", {
                target: {
                    value: "1",
                },
            });
            await wrapper.update();
            expect(wrapper.find("#pidRangeFrom").instance().value).toEqual("1");
            wrapper.find("#pidRangeTo").simulate("change", {
                target: {
                    value: "5",
                },
            });
            await wrapper.update();
            expect(wrapper.find("#pidRangeTo").instance().value).toEqual("5");
            wrapper.find("button").simulate("click");
        });

        await Promise.resolve();

        expect(fetchContextValues.action.fetchText).toHaveBeenCalledWith(
            expect.stringMatching(/messenger\/queuesolrindex/),
            {
                method: "POST",
                body: '{"prefix":"foo:","to":"5","from":"1"}',
            },
            { "Content-Type": "application/json" },
        );
        expect(setResults).toHaveBeenCalledWith("testText");
    });

    it("throws an error on api call", async () => {
        fetchContextValues.action.fetchText.mockRejectedValue({
            message: "testError",
        });

        await act(async () => {
            wrapper.find("button").simulate("click");
        });

        await Promise.resolve();

        expect(fetchContextValues.action.fetchText).toHaveBeenCalled();
        expect(setResults).toHaveBeenCalledWith("testError");
    });
});
