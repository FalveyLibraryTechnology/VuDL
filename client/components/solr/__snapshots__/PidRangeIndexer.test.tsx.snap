// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PidRangeIndexer renders 1`] = `
<PidRangeIndexer
  setResults={[MockFunction]}
>
  <label>
    Prefix:
    <input
      id="pidRangePrefix"
      onChange={[Function]}
      type="text"
      value="vudl:"
    />
  </label>
  <label>
    From (number):
    <input
      id="pidRangeFrom"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label>
    To (number): 
    <input
      id="pidRangeTo"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    onClick={[Function]}
  >
    Queue Range
  </button>
</PidRangeIndexer>
`;
